### PR TITLE
MNT: Update register_transform for IP 7

### PR DESCRIPTION
**Pull request by tacaswell**
Also adds a test for both IPthon APIs

This should be backported to the 1.5.x branch

## Description

Fixes the `register_transform` for ip7

## Motivation and Context

It was broken due to changes in the API of IPyhton

## How Has This Been Tested?

Added tests.

### CODE REVIEW

File: bluesky/utils.py [hunk 1]
```diff
@@ -894,36 +894,77 @@ def update_sub_lists(out, inp):
...
+            if line.startswith(prefix):
+                line = line[len(prefix):].strip()
+                return '{}({})'.format(RE, line)
+            return line
+
+        ip.input_splitter.logical_line_transforms.append(tr_re())
+        ip.input_transformer_manager.logical_line_transforms.append(tr_re())
+
+
+class PlanCompleter:
```
Comment (danielballan) [inline bluesky/utils.py]: This isn’t mentioned in the PR description. Can you add a docstring?

File: bluesky/utils.py [hunk 2]
```diff
@@ -894,36 +894,77 @@ def update_sub_lists(out, inp):
...
+
+
+def register_transform_complete(RE_name, complete_targets, *, prefix='<', ip=None):
+    import IPython
+
+    if ip is None:
+        ip = IPython.get_ipython()
+
+    register_transform(RE_name, prefix=prefix, ip=ip)
+    pc = PlanCompleter(['count', 'scan'])
```
Comment (danielballan) [inline bluesky/utils.py]: Can you comment on why these two are hard-coded?

**Comment by tacaswell** [comment]
Removed the WIP commit that I should have pushed to a different branch made the string version checking the same everywhere.

File: bluesky/tests/test_transformer.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    assert cell == new_cell
+
+
+@pytest.mark.parametrize('cell',
+                         ['< b\n',
+                          '<a\n',
+                          '<fly(a, b, c), md={"a": "aardvark"}\n',
+                          '  <b\n',
+                          '<a, c=d\n'])
+def test_nransform(transform_cell, cell):
```
Comment (mrakitin) [inline bluesky/tests/test_transformer.py]:
```suggestion
def test_transform(transform_cell, cell):
```

File: bluesky/tests/test_transformer.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    new_cell = transform_cell(cell)
+    assert cell == new_cell
+
+
+@pytest.mark.parametrize('cell',
+                         ['< b\n',
+                          '<a\n',
+                          '<fly(a, b, c), md={"a": "aardvark"}\n',
+                          '  <b\n',
+                          '<a, c=d\n'])
```
Comment (mrakitin) [inline bluesky/tests/test_transformer.py]: Is inconsistent usage of blank spaces intended here?

Comment (tacaswell) [reply bluesky/tests/test_transformer.py]: Yes, it is testing that it still works if we get wonky input from the user.

Comment (mrakitin) [reply bluesky/tests/test_transformer.py]: That's a reasonable point.

**Comment by tacaswell** [comment]
backported to v1.5.x as 0513943b3beb71eb107d57be56ad342855dec02d

File: bluesky/tests/test_transformer.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import pytest
+from bluesky.utils import register_transform
+
+@pytest.fixture
+def transform_cell():
+    IPython = pytest.importorskip('IPython')
+    ip = IPython.core.interactiveshell.InteractiveShell()
+    register_transform('RE', prefix='<', ip=ip)
+    if IPython.__version__ > '7':
```
[the remark this answers is not in the document]
Comment (danielballan) [reply bluesky/tests/test_transformer.py]: Re-upping this:

```python
In [1]: import bluesky

In [2]: bluesky.__version__
Out[2]: '1.5.5'

In [3]: import IPython

In [4]: IPython.__version__
Out[4]: '6.5.0'

In [5]: from bluesky.utils import register_transform

In [6]: from bluesky import RunEngine

In [7]: RE = RunEngine()

In [8]: register_transform(RE)

In [9]: from bluesky.plans import  count

In [10]: <count([])
  File "<ipython-input-10-ff0ead2ea055>", line 1
    <bluesky.run_engine.RunEngine object at 0x7f2e20b84128>(count([]))
    ^
SyntaxError: invalid syntax
```

[the remark this answers is not in the document]
Comment (tacaswell) [reply bluesky/tests/test_transformer.py]: `register_transform` takes the string `'RE'` not the `RE` object.



```python
amilyn@10:34  ➤  ipython
Python 3.7.3 (default, Mar 27 2019, 22:11:17) 
Type 'copyright', 'credits' or 'license' for more information
IPython 6.5.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import IPython

In [2]: IPython.__version__
Out[2]: '6.5.0'

In [3]: import bluesky

In [4]: bluesky.__version__
Out[4]: '1.5.5'

In [5]: bluesky.utils.register_transform('print')

In [6]: <'hi'
hi

In [7]: 


```

[the remark this answers is not in the document]
Comment (danielballan) [reply bluesky/tests/test_transformer.py]: It turns out `register_transform` wants a name in the global namespace, so `'RE'` not `RE`.

**Pull request merged**